src/huggingface_hub/hf_api.py

```diff
@@ -212,6 +212,57 @@ def __post_init__(self):
             self.pr_num = None
 
 
+class RepoUrl(str):
+    """Subclass of `str` describing a repo URL on the Hub.
+
+    Inherits from `str` for backward compatibility.
+
+    RepoUrl adds new properties:
+    - endpoint (`str`)
+    - namespace (`Optional[str]`)
+    - repo_id (`str`)
+    - repo_type (`Literal["model", "dataset", "space"]`)
+    - url (`str`)
+
+    `RepoUrl` is returned by `HfApi.create_repo`.
+
+    Example:
+    ```py
+    >>> RepoUrl('https://huggingface.co/gpt2')
+    RepoUrl('https://huggingface.co/gpt2', endpoint='https://huggingface.co', repo_type='model', repo_id='gpt2')
+
+    >>> RepoUrl('https://hub-ci.huggingface.co/dataset/dummy_user/dummy_dataset', endpoint='https://hub-ci.huggingface.co')
+    RepoUrl('https://hub-ci.huggingface.co/dataset/dummy_user/dummy_dataset', endpoint='https://hub-ci.huggingface.co', repo_type='dataset', repo_id='dummy_user/dummy_dataset')
+
+    >>> HfApi.create_repo("dummy_model")
+    RepoUrl('https://huggingface.co/Wauplin/dummy_model', endpoint='https://huggingface.co', repo_type='model', repo_id='Wauplin/dummy_model')
+    ```
+    """
+
+    def __new__(cls, content: Any, endpoint: Optional[str] = None):
+        return super(RepoUrl, cls).__new__(cls, content)
+
+    def __init__(self, content: Any, endpoint: Optional[str] = None) -> None:
```

Comment:
Should we document these params in the class docstring Args?

Reply:
I hesitated to do so but then thought this class this not really meant to be instantiated by anyone except in `huggingface_hub`. I made the documentation more focused on how to use an object of this class instead.
But I could be convinced if you think it can prove useful :)

Reply:
No strong opinion, let's see what others think :)

Reply:
@osanseviero In the end, I've added the docstring for args so that it's done.

Reply:
Thanks!